fn_falcon_sandbox/fn_falcon_sandbox/components/falcon_sandbox_submit_file.py

```diff
@@ -0,0 +1,229 @@
+                        "Error: [App Config] Minimum value for fetch_report_timeout and fetch_report_status_interval should be 60."
+                    )
+            else:
+                raise FunctionError(
+                    "Error: [App Config] fetch_report_timeout and fetch_report_status_interval cannot left empty."
+                )
+            ## Checking and preparing inputs
+            if falcon_sandbox_incident_id is None:
+                raise FunctionError(
+                    "Error: falcon_sandbox_incident_id must be specified."
+                )
+            elif (
+                falcon_sandbox_attachment_id is None
+                and falcon_sandbox_artifact_id is None
+            ):
+                raise FunctionError(
+                    "Error: falcon_sandbox_attachment_id or falcon_sandbox_artifact_id must be specified."
+                )
+            else:
+                yield StatusMessage("> Function inputs OK")
+
+            falcon_sandbox_environment_id = get_environment_id(
+                falcon_sandbox_environment
+            )
+            falcon_sandbox_action_script = get_runtime_action_script(
+                falcon_sandbox_action_script
+            )
+
+            ## Making sure all param names are following naming conventions
+            for p in kwargs:
+                if not "falcon_sandbox_" in p:
+                    FunctionError(
+                        "Input parameter name must start with 'falcon_sandbox_'"
+                    )
+
+            client = self.rest_client()
+            time_now = str(time.time())
+            request_common = RequestsCommon(self.options, kwargs)
+
+            ## Get the file for submission
+            yield StatusMessage(
+                "Getting Attachment from Incident {}".format(falcon_sandbox_incident_id)
+            )
+
+            attachment, attachment_metadata = get_file_attachment_and_metadata(
+                client,
+                falcon_sandbox_incident_id,
+                falcon_sandbox_artifact_id,
+                falcon_sandbox_task_id,
+                falcon_sandbox_attachment_id,
+            )
+
+            attachment_path, temp_files = write_temp_file(
+                attachment, "attachment_{}".format(time_now)
+            )
+
+            attachment_name = attachment_metadata.get("name")
+            attachment_type = attachment_metadata["content_type"]
+
+            yield StatusMessage(
+                "Environment using for analysis: {}".format(falcon_sandbox_environment)
+            )
+
+            ## Prepare for submission
+            url, http_method = HA_REST_API_URLS.get("submit_file")
+            url = url.format(API_HOST)
+            form_data = {
+                "file": (attachment_name, open(attachment_path, "rb"), attachment_type)
+            }
+
+            for p in HA_LIST_OF_RUNTIME_PARAMS_SUBMIT_FILE:
+                k = p.split("falcon_sandbox_")[1]
+                v = eval(p)
```

Comment:
Can you explain what happens here with `eval`?

Reply:
Sure,
I made a strict convention here for Falcon sandbox's runtime parameter names i.e. parameter's actual name followed by `'falcon_sandbox_'`, e.g. `falcon_sandbox_no_share_third_party` corresponds to actual API's name `no_share_third_party`. Which we need to pass with the Form data like:  
`{"no_share_third_party": "falcon_sandbox_no_share_third_party", ....}`  
Reason for so is to avoid writing repeated code